mem0/vector_stores/pgvector.py

```diff
@@ -0,0 +1,257 @@
+import json
+from typing import Optional, List, Dict, Any
+from pydantic import BaseModel
+
+try:
+    import psycopg2
+    from psycopg2.extras import execute_values
+except ImportError:
+    raise ImportError("PGVector requires extra dependencies. Install with `pip install psycopg2`") from None
+
+
+from mem0.vector_stores.base import VectorStoreBase
+
+class OutputData(BaseModel):
+    id: Optional[str]
+    score: Optional[float]
+    payload: Optional[dict]
+
+
+class PGVector(VectorStoreBase):
+    def __init__(
+        self,
+        dbname,
+        collection_name,
+        embedding_model_dims,
+        user,
+        password,
+        host,
+        port
+    ):
+        """
+        Initialize the PGVector database.
+
+        Args:
+            dbname (str): Database name
+            collection_name (str): Collection name
+            embedding_model_dims (int): Dimension of the embedding vector
+            user (str): Database user
+            password (str): Database password
+            host (str, optional): Database host
+            port (int, optional): Database port
+        """
+        self.conn = psycopg2.connect(
+            dbname=dbname,
+            user=user,
+            password=password,
+            host=host,
+            port=port
+        )
+        self.cur = self.conn.cursor()
+
+        self.create_col(collection_name, embedding_model_dims)
+    
+    def create_col(self, name, embedding_model_dims):
+        """
+        Create a new collection (table in PostgreSQL).
+
+        Args:
+            name (str): Name of the collection.
+            embedding_model_dims (int, optional): Dimension of the embedding vector.
+        """
+        self.cur.execute(f"""
+            CREATE TABLE IF NOT EXISTS {name} (
+                id UUID PRIMARY KEY,
+                vector vector({embedding_model_dims}),
+                payload JSONB
```

Comment:
Do we need the payload column? Why not create column for the keys in the payload since most of the fields are going to be standard anyways.

Reply:
Honestly it was easy to use payload directly :) and I didn't see any benefit for creating columns for individual keys.

Reply:
What are the keys inside the payload column? Based on that, we may have to add columns here.

Reply:
data, user_id, agent_id, run_id, metadata, filters. And the reason to directly use payload is we have the same flow for other vectordbs as well.